tests/invalid_models_tests/test_ordinary_fields.py

```diff
@@ -845,6 +850,43 @@ class Model(models.Model):
                     ],
                 )
 
+    def test_non_iterable_choices(self):
+        class Model(models.Model):
+            field = models.IntegerField(choices=123)
+
+        field = Model._meta.get_field("field")
+        self.assertEqual(
+            field.check(),
+            [
+                Error(
+                    "'choices' must be a mapping (e.g. a dictionary) or an iterable "
+                    "(e.g. a list or tuple).",
+                    obj=field,
+                    id="fields.E004",
+                ),
+            ],
+        )
+
+    def test_non_iterable_choices_number(self):
+        """An integer isn't a valid choice pair."""
+
+        class Model(models.Model):
+            field = models.IntegerField(choices=[123])
+
+        field = Model._meta.get_field("field")
+        self.assertEqual(
+            field.check(),
+            [
+                Error(
+                    "'choices' must be a mapping of actual values to human readable "
+                    "names or an iterable containing (actual value, human readable "
+                    "name) tuples.",
+                    obj=field,
+                    id="fields.E005",
+                ),
+            ],
+        )
```

Comment:
Extra test coverage should be moved to a separate commit/PR.

Reply:
Sorry, I've run out of time to fiddle this around as the error message would be different.

Reply:
Moved to the #17152.